### DISCUSSION TITLE
.net 6.0 container pods running in k8 cluster using a lot of memory.

### DISCUSSION THREAD

**Discussion by haristauqir** [dotnet/dotnet-docker#4546] [Q&A]
Hi,

We have 30 microservices deployed in k8 cluster and some services which are sitting in the cluster and doing nothing are using somewhere between 150-200 MB. The other ones which are CRUD API services are taking somewhere between 200 - 500 MB. I tried adding memory to one of the pod to 700mb and it started using 650 mb. I wanted to ask shall i change my GC to workstation? Any ideas.???

**Comment by akoken** [comment]
Yes! I highly recommend using [Workstation GC](https://learn.microsoft.com/en-us/aspnet/core/performance/memory?view=aspnetcore-3.0#gc-using-docker-and-small-containers) for your microservices running on containers. However, before making any changes, it's important to check if your application has any memory leaks. You can easily obtain a memory dump by attaching to your pod if you are not using distroless linux.

To attach to the pod, you can use the following command:
```bash
kubectl exec -ti <pod-name> -- /bin/sh
```
Once you have attached to the pod, you can obtain a memory dump using the following commands:
```bash
# List all the current processes where a garbage collector dump can be obtained from
./dotnet-gcdump ps 

# This command will create a dump for the given PID (in this case, 1)
./dotnet-gcdump collect -p 1 

# Download the dump file from your pod to your local machine
kubectl cp <pod-name>:/tools/dotnet_20230331_163848.gcdump .\result.gcdump
```
Please note that `dotnet-gcdump` does not exist in the runtime image, so you will need to install it by using the following command in your build stage and then copy it to your final runtime image:

``` bash
dotnet tool install --global dotnet-gcdump
```
I hope this helps!
